### ISSUE TITLE
`Addresses` table + model semantics - Child of #3779

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#5326]
Child of #3779
Prerequisite: #5246

## Description

Update the Addresses table and model to replace `Chain` references with `Community` references.

## Project Owner

@timolegros 

## Engineering Requirements

- [x] Rename the `chain` column to `community_id`
- [x] Rename any foreign keys or indexes associated with the column
- [x] Update all direct (non-variable) references to the `Address.chain` sequelize property

**Comment by timolegros** [comment]
Increasing points to (2) since there are tons of references that need to be reviewed individually.

**Comment by timolegros** [comment]
Surface area keeps increasing so bumping again. Need to be very thorough in testing all affected features.

**Comment by timolegros** [comment]
Required CI is passing and most bugs should be resolved (one remaining obvious profile-related bug -> failing E2E test). Will do one last manual testing of all affected features tomorrow before putting the PR up for review. Will note that the PR must be merged as quickly as possible after it is ready because every time another PR is merged into master I need to review every change in that PR to ensure there are no indirect references to `address.chain` that aren't caught in CI (has occurred numerous times now).

**Comment by timolegros** [comment]
Blocked by #5451 since the fix is required to test Discobot route updates.

**Comment by timolegros** [comment]
Bumped one last time as the woes continued. Manual testing revealed several DiscordBot issues which block proper testing of the route changes.
